### PR TITLE
Fix challenge errors related to allocation expiry and partition 

### PR DESCRIPTION
**Pull request by Jayashsatolia403**
## Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/0chain/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

## Associated PRs (Link as appropriate):
- blobber:
- gosdk:
- system_test:
- zboxcli:
- zwalletcli:
- Other: ...

### CODE REVIEW

**Review by peterlimg — APPROVED** [review]
LGTM

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | false  |
| system-tests  | master  |
| gosdk  | staging  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | fix/blobber_rewards  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0dns  | staging  |

File: code/go/0chain.net/smartcontract/storagesc/challenge.go [hunk 1]
```diff
@@ -434,6 +442,8 @@ func (sc *StorageSmartContract) verifyChallenge(t *transaction.Transaction,
 		errCode   = "verify_challenge"
 	)
 
+	fmt.Println("verifyChallenge", balances.GetBlock().Round)
```
Comment (peterlimg) [inline code/go/0chain.net/smartcontract/storagesc/challenge.go]: Remove if it's for local debugging or use `logger`

File: code/go/0chain.net/smartcontract/storagesc/challenge.go [hunk 2]
```diff
@@ -626,8 +636,13 @@ func (sc *StorageSmartContract) challengePassed(
 
 	rewardRound := GetCurrentRewardRound(balances.GetBlock().Round, triggerPeriod)
 	// this expiry of blobber needs to be corrected once logic is finalized
-	if blobber.RewardRound.StartRound != rewardRound ||
-		balances.GetBlock().Round == 0 {
+
+	fmt.Println("Code round : ", balances.GetBlock().Round)
+
+	fmt.Println("Reward round : ", rewardRound)
+	fmt.Println("Blobber reward round : ", blobber.RewardRound.StartRound)
```
Comment (peterlimg) [inline code/go/0chain.net/smartcontract/storagesc/challenge.go]: Use logger or remove if it's for debugging

**Pull request merged**